### PR TITLE
EntityKeyValidator: fixed bug, nodes not validated on sibling entity delete

### PR DESCRIPTION
**Pull request by fefrain**
…delete

### CODE REVIEW

File: core/record/_recordValidator/attributeKeyValidator.js [hunk 1]
```diff
@@ -19,19 +18,13 @@ const validateAttributeKey = (survey, record, attributeDef) => async (propName,
 
 const _isEntityDuplicate = (survey, record, entity) => {
   // 1. get sibling entities
-  const parentNode = Record.getParentNode(entity)(record)
-  const siblingEntitiesAndSelf = Record.getNodeChildrenByDefUuid(parentNode, Node.getNodeDefUuid(entity))(record)
-  const siblingEntities = R.reject(R.propEq(Node.keys.uuid, Node.getUuid(entity)), siblingEntitiesAndSelf)
+  const siblingEntities = Record.getNodeSiblings(entity)(record)
```
Comment (minotogna) [inline core/record/_recordValidator/attributeKeyValidator.js]: does this return all siblings or only sibling entities?

Comment (fefrain) [reply core/record/_recordValidator/attributeKeyValidator.js]: the node passed to getNodeSiblings is an entity -> its siblings are entities too

Comment (minotogna) [reply core/record/_recordValidator/attributeKeyValidator.js]: why? a tree, inside a plot, cannot a have sibling like plot no? plot location?

Comment (fefrain) [reply core/record/_recordValidator/attributeKeyValidator.js]: these are not siblings but children...

Comment (fefrain) [reply core/record/_recordValidator/attributeKeyValidator.js]: you are right, I'll add a filter function then

**Pull request merged**